senders/msteams/datatypes.go

```diff
@@ -105,3 +109,15 @@ type MessageCard struct {
 	Sections        []Section `json:"sections"`
 	PotentialAction []Actions `json:"potentialAction,omitempty"`
 }
+
+type ErrTeamsError struct {
+	error       string
+	description string
+	err         error
+}
+
+func (e *ErrTeamsError) Error() string {
+	return fmt.Sprintf("%s : %s", e.description, e.error)
+}
+
+func (e *ErrTeamsError) Unwrap() error { return e.err }
```

Comment:
Hey @imavroukakis there is no need to implement this error structures by yourself. My comment was only about changing format string to use `%w` placeholders instead of `%s`. The default errors returned by `errors.New` and `fmt.Errorf` are compatible with Go 1.13 error values. You can check example here https://play.golang.org/p/_2gKhr5mt05

Let's just remove this struct and use built-in errors generated by `errors.New` and `fmt.Errorf`

Reply:
So, for returning an error with a status code, this is acceptable?
```
if response.StatusCode >= http.StatusBadRequest && response.StatusCode <= http.StatusNetworkAuthenticationRequired {
		return fmt.Errorf("server responded with a non 2xx code: %d", response.StatusCode)

}
```

Reply:
Yeah, this is enough. If you will handle this error somewhere on higher levels, you will be able just wrap it with `fmt.Errorf("some additional info: %w")`

Another one note: Please check that all errors do not contains capitalised strings https://github.com/golang/go/wiki/CodeReviewComments#error-strings